packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,197 @@
+        liveness = _liveness;
+    }
+
+    function proposeTransactions(Transaction[] memory transactions, bytes memory ancillaryData) public {
+        // create a proposal with a bundle of transactions
+        // note: based in part on the UMA Governor contract
+        // https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/implementation/Governor.sol
+        // note: optional ancillaryData explains the intent of the transactions to make comprehension easier
+        uint256 id = proposals.length;
+        uint256 time = block.timestamp;
+        address proposer = msg.sender;
+
+        // Add a zero-initialized element to the proposals array.
+        proposals.push();
+
+        // Initialize the new proposal.
+        Proposal storage proposal = proposals[id];
+        proposal.requestTime = time;
+        proposal.ancillaryData = ancillaryData;
+
+        // Initialize the transaction array.
+        for (uint256 i = 0; i < transactions.length; i++) {
+            require(transactions[i].to != address(0), "The `to` address cannot be 0x0");
+            // If the transaction has any data with it the recipient must be a contract, not an EOA.
+            if (transactions[i].data.length > 0) {
+                require(_isContract(transactions[i].to), "EOA can't accept tx with data");
+            }
+            proposal.transactions.push(transactions[i]);
+        }
+
+        // Propose a set of transactions to the OO. If not disputed, they can be executed with executeProposal().
+        // docs: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/interfaces/SkinnyOptimisticOracleInterface.sol
+        optimisticOracle.requestAndProposePriceFor(
+            identifier,
+            uint32(time),
+            ancillaryData,
+            collateral,
+            0,
+            bond,
+            liveness,
+            proposer,
+            // Canonical value representing "True"; i.e. the transactions are valid.
+            int256(1e18)
+        );
+
+        emit TransactionsProposed(id, proposer, time);
+    }
+
+    function executeProposal(
+        uint256 _proposalId,
+        uint256 _transactionIndex,
+        Enum.Operation _operation
+    ) public payable {
+        // execute transactions in an approved proposal using exec() function
+        Proposal storage proposal = proposals[_proposalId];
+
+        // this will revert if the price has not settled
+        int256 price = _getOracle().getPrice(identifier, proposal.requestTime, proposal.ancillaryData);
+
+        Transaction memory transaction = proposal.transactions[_transactionIndex];
+
+        require(
+            _transactionIndex == 0 || proposal.transactions[_transactionIndex - 1].to == address(0),
+            "Previous tx not yet executed"
+        );
+        require(transaction.to != address(0), "Tx already executed");
+        require(price != 0, "Proposal was rejected");
```

Comment:
Can we delete a proposal that was rejected to reduce storage costs and get a refund?

Reply:
Yep!